src/battle-log.ts

```diff
@@ -306,37 +307,40 @@ class BattleLog {
 			this.prependDiv('notice uhtml-' + id, BattleLog.sanitizeHTML(html));
 		}
 	}
-	hideChatFrom(userid: ID, showRevealButton = true) {
+	hideChatFrom(userid: ID, showRevealButton = true, lineCount = 0) {
 		const classStart = 'chat chatmessage-' + userid + ' ';
-		let lastNode;
-		let count = 0;
-		for (const node of this.innerElem.childNodes as any) {
+		let nodes: HTMLElement[] = [];
+		for (const node of this.innerElem.childNodes as any as HTMLElement[]) {
 			if (node.className && (node.className + ' ').startsWith(classStart)) {
-				node.style.display = 'none';
-				node.className = 'revealed ' + node.className;
-				count++;
+				nodes.push(node);
 			}
-			lastNode = node;
 		}
 		if (this.preemptElem) {
-			for (const node of this.preemptElem.childNodes as any) {
+			for (const node of this.preemptElem.childNodes as any as HTMLElement[]) {
 				if (node.className && (node.className + ' ').startsWith(classStart)) {
-					node.style.display = 'none';
-					node.className = 'revealed ' + node.className;
-					count++;
+					nodes.push(node);
 				}
-				lastNode = node;
 			}
 		}
-		if (!count || !showRevealButton) return;
+		if (lineCount) nodes = nodes.slice(-lineCount);
+
+		for (const node of nodes) {
+			if (node.className && (node.className + ' ').startsWith(classStart)) {
+				node.style.display = 'none';
+				node.className = 'revealed ' + node.className;
+			}
```

Comment:
```suggestion
			node.style.display = 'none';
			node.className = 'revealed ' + node.className;
```

Whoopsie, the `if` block is redundant here.